blog/_posts/2018-06-13-scala-213-collections.md

```diff
@@ -0,0 +1,210 @@
+is traversed.
+
+Furthermore, we have introduced an operation named `groupMap`
+that both groups elements and transforms the groups. The above code
+can be rewritten as follows to take advantage of `groupMap`:
+
+~~~
+def namesByAge(users: Seq[User]): Map[Int, Seq[String]] =
+  users.groupMap(_.age)(_.name)
+~~~
+
+The returned `Map` is strict: it eagerly evaluates its elements
+once. Also, the fact that it is implemented as a single operation
+makes it possible to apply some optimizations that make it
+~1.3x faster than the version that uses `mapValues`.
+
+### `InPlace` Transformation Operations
+
+Mutable collections have a couple of new operations for transforming
+their elements in place: instead of returning a new collection (like
+`map` and `filter` do) they mutate the source collection. These
+operations are suffixed with `InPlace`. For instance, to remove
+users whose name start with the letter `J` from a buffer and then
+increment their age, one can now write:
+
+~~~ scala
+val users = ArrayBuffer(…)
+users
+  .filterInPlace(user => !user.name.startsWith("J"))
+  .mapInPlace(user => user.copy(age = user.age + 1))
+~~~
+
+## Deprecations For Less Confusion
+
+A consequence of cleaning and simplifying the collections framework
+is that several types or operations have been deprecated.
+
+### `Iterable` Is The Top Collection Type
+
+We felt that having a distinction between `Traversable` and `Iterable` was not
+worth it, so we removed `Traversable` (it is now an alias to `Iterable[A]`).
+
+`IterableOnce[A]` is now the collection type at the top of the hierarchy.
+Its only abstract member is `def iterator: Iterator[A]`.
+
+### `LazyList` Is Preferred Over `Stream`
+
+`Stream` is deprecated in favor of `LazyList`. As its name suggests,
+a `LazyList` is a linked list whose elements are lazily evaluated. An
+important semantic difference with `Stream` is that in `LazyList` both
+the head and the tail are lazy, whereas in `Stream` only the tail is lazy.
+
+### Insertion And Removal Operations Are Not Available On Generic Collections
+
+In the current framework, the `scala.collection.Map` type has a `+` and a `-` operations
+to add and remove entries. The semantics of these operations is to return a new collection
+with the added or removed entries, without changing the source collection.
+
+These operations are then inherited by the mutable branch of the collections. But the mutable
+collection types also introduce their own insertion and removal operations, namely `+=` and `-=`,
+which modify the source collection in place. This means that the `scala.collection.mutable.Map` type
+has `+` and `+=`, as well as `-` and `-=`.
+
+Having all these operations can be handy in some cases but can also introduce confusion. If you want
+to use `+` or `-`, then you probably wanted to use an immutable collection type in the first place…
+Another example is the `updated` operation, which is available on mutable `Map` but returns a new
+collection.
```

Comment:
Can you add a summarizing sentence saying exactly which methods were deprecated from which entities here? Just so it's as clear as possible for anyone skimming.

Reply:
That would be something like: “`+`, `-`, `updated` have been moved from `scala.collection.Map` to `scala.collection.immutable.Map`, and `+` and `-` have been moved from `scala.collection.Set` to `scala.collection.immutable.Set`”. Should I replace the sentence line 194 with that?